src/libcrun/utils.c

```diff
@@ -2183,3 +2183,32 @@ has_suffix (const char *str, const char *suffix)
     return 0;
   return memcmp (str + lenstr - lensuffix, suffix, lensuffix) == 0;
 }
+
+// Note: str_join_array expects array to start from index 1.
```

Comment:
this is some strange requirement.

Could we pass `offset` as an argument?  For the wasmer case you'll set it to `1`

Reply:
yes makes sense.